src/Orleans/Cancellation/GrainCancellationToken.cs

```diff
@@ -56,16 +61,19 @@ public sealed class GrainCancellationToken : IDisposable
 
         internal Task Cancel()
         {
-            _cancellationTokenSource.Cancel();
+            // propagate the exception from the _cancellationTokenSource.Cancel back to the caller
+            // but also cancel _targetGrainReferences. 
+            Task task = OrleansTaskExtentions.WrapInTask(_cancellationTokenSource.Cancel);
```

Comment:
Ah, I see. Sure then, I guess doing a try catch and aggregating exceptions will make the code harder to read in this case. I'm fine keeping it as is, thanks for the clarification.